### PR TITLE
ci: put failing tests into e2e job summary

### LINKED ISSUE #47444: add nice summary of failing tests in github

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/47444

### Description

Add job summary to simplify search of the failing tests, so far solution is pretty straightforward, later on we can combine reports from different jobs into one using mochawesome-merge

### How to verify

check the failing job Summary page https://github.com/metabase/metabase/actions/runs/10629389381?pr=47134
### Demo

image

### CODE REVIEW

**Comment by camsaul** [comment]
What is this supposed to look like? This is what I see

image

**Comment by uladzimirdev** [comment]
nope, @camsaul. it's not ready for review 😅

File: .github/scripts/handle-mochawesome-report.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
+const fs = require("fs");
+
+const logger = { debug: console.debug, error: console.error };
+
+const collectTestsByType = ({ type, suite, path, cache = [] }) => {
+  logger.debug(`Collecting tests of type ${type}`);
+  const localCache = cache;
+  const { [type]: typeList, suites, tests } = suite;
+
+  if (typeList.length > 0) {
```
Comment (kamilmielnik) [inline .github/scripts/handle-mochawesome-report.js:10]: Redundant condition

File: .github/scripts/handle-mochawesome-report.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        logger.error(`Test with uuid ${uuid} not found`);
+        throw new Error(`Test with uuid ${uuid} not found`);
+      }
+      logger.debug(`Found test with uuid ${uuid}`);
+      foundTestByUuid.path = path;
+      localCache.push({ path, ...foundTestByUuid });
+    }
+  }
+
+  if (suites.length > 0) {
```
Comment (kamilmielnik) [inline .github/scripts/handle-mochawesome-report.js:23]: Redundant condition

File: .github/scripts/handle-mochawesome-report.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+  let summary = "### Failed Tests Summary\n\n";
+
+  for (const [path, tests] of Object.entries(failedTestsByPath)) {
+    summary += `| ${path} |\n`;
+    summary += "| :--- |\n";
+
+    tests.forEach(test => {
+      summary += `| ${test.title}`;
+      summary += "<p></p>"; // adds extra space between title and
```
Comment (kamilmielnik) [inline .github/scripts/handle-mochawesome-report.js:96]:
```suggestion
      summary += "<p></p>"; // adds extra space after title
```

File: .github/scripts/handle-mochawesome-report.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
+const fs = require("fs");
+
+const logger = { debug: console.debug, error: console.error };
+
+const collectTestsByType = ({ type, suite, path, cache = [] }) => {
+  logger.debug(`Collecting tests of type ${type}`);
+  const localCache = cache;
```
Comment (kamilmielnik) [inline .github/scripts/handle-mochawesome-report.js:7]: Mutating an argument is not great but since it's a small standalone script... :see_no_evil:

**Review by kamilmielnik — APPROVED** [review]
I can't wait for my tests to fail with these nice summaries :partying_face:

**Review by iethree — APPROVED** [review]
looks nice!

**Pull request merged**